pkg/apiclient/application/forwarder_overwrite.go

```diff
@@ -115,9 +115,11 @@ func init() {
 	logsForwarder := func(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w gohttp.ResponseWriter, req *gohttp.Request, recv func() (proto.Message, error), opts ...func(context.Context, gohttp.ResponseWriter, proto.Message) error) {
 		if req.URL.Query().Get("download") == "true" {
 			w.Header().Set("Content-Type", "application/octet-stream")
+			appNamespace := req.URL.Query().Get("appNamespace")
+			podName := req.URL.Query().Get("podName")
 			fileName := "log"
-			if container := req.URL.Query().Get("container"); len(container) > 0 && kube.IsValidResourceName(container) {
-				fileName = container
+			if kube.IsValidResourceName(appNamespace) && kube.IsValidResourceName(podName) {
```

Comment:
```suggestion
			if kube.IsValidResourceName(appNamespace) && kube.IsValidResourceName(podName) {
```

`appNameswpace` is argocd always (until you change the namespace where argocd has been installed). Instead, the namespace should be the one that the pod is part of. Furthermore, testing the above changes will give me a filename as `argocd-my-app-dev-798dbbd986-kc2vz.log`. If my pod wasn't part of deployment, I'd want to have log file in this format - `<namespace that pod is part of>-<pod-name>`.